### ISSUE TITLE
fftwf_plan_many_dft_r2c causes segmentation error

### ISSUE THREAD

**Issue by kuangdai** [FFTW/fftw3#233]
I triggered the error with the following code:

```c++
extern "C" {
#include <fftw3.h>
}

int main() {
    // max size
    int maxNR = 51;
    int maxNC = maxNR / 2 + 1;
    int HOWMANY = 150;
    int NR = 20;
    
    // data allocation by fftw
    auto *rdata = fftw_alloc_real(maxNR * HOWMANY);
    auto *cdata = fftw_alloc_complex(maxNC * HOWMANY);
    
    // create plan
    fftw_plan plan = fftw_plan_many_dft_r2c(1, &NR, HOWMANY,
                                            rdata, NULL, 1, maxNR,
                                            cdata, NULL, 1, maxNC,
                                            FFTW_PATIENT);
    
    // free
    fftw_destroy_plan(plan);
    fftw_free(rdata);
    fftw_free(cdata);
}
```

To trigger this error, the maximum logical size (`maxNR`) must be an odd number and the logical size (`NR`) must be some specific numbers: I only found 20 and 24. 

I tested with g++ 7/8/9 on Linux and clang 12 on macOS. I don't think the error is compiler dependent.

**Comment by kuangdai** [comment]
Forgot to mention: the flag must be `FFTW_PATIENT` to trigger the error.

**Comment by matteo-frigo** [comment]
Looks like you found a real bug, thank you.  Should be fixed in 06855286.
